shacl12-inf-rules/index.html

```diff
@@ -160,444 +223,1880 @@
+          </dd>
+
+          <dt><dfn>condition expression</dfn></dt>
+          <dd>
+            A <em>condition expression</em> is a function, or functional form,
+            that evaluates to true or false.
+            [=Condition expressions=] appear in the [=body=] of a [=rule=].
+          </dd>
+
+          <dt><dfn>assignment</dfn></dt>
+          <dd>
+            An <em>assignment</em> is a pair of a <a>variable</a>, called the
+            <dfn>assignment variable</dfn>, and an <a>expression</a>, called the
+            <dfn>assignment expression</dfn>.
+            [=Assignments=] appear in the [=body=] of a [=rule=].
+          </dd>
+
+          <dt><dfn>rule</dfn></dt>
+          <dd>
+            A <em>rule</em> is a pair of a [=rule head=] (often just "head") and
+            a [=rule body=] (often just "body").
+          </dd>
+
+          <dt><dfn data-lt="head|rule head">rule head</dfn></dt>
+          <dd>
+            A <em>rule head</em> is a sequence where each element
+            of the sequence is a [=triple template=].
+          </dd>
+          
+          <dt>
+            <dfn data-lt="body|rule body">rule body</dfn>
+          </dt>
+          <dd>
+            A <em>rule body</em> is a sequence where each element
+            of the sequence is a [=triple pattern=], a [=condition expression=] or an [=assignment=].
+          </dd>
+          
+          <dt><dfn>rule set</dfn></dt>
+          <dd>
+            A <em>rule set</em> is a collection of zero or more rules.
+          </dd>
+        </dl>
+
+        <p>
+          In a [=triple pattern=] or a [=triple template=],
+          position 1 of the tuple is informally called the <em>subject</em>;
+          position 2 is informally called the <em>predicate</em>
+          position 3 is informally called the <em>object</em>.
+        </p>
+
+        <section id="wellformed">
+          <h3>Well-formedness Conditions</h3>
+          <p>
+            Well-formedness is a set of conditions on abstract syntax of
+            shapes rules.  Together, these rules ensure that a [=variable=] in a
+            [=head=] of a rule has a value defined in the [=body=] of the rule; 
+            that each variable in an condition expression or assignment 
+            expression has a value at the point of evaluation, and that each
+            assignment in a rule only introduces a new variable, 
+            not used earlier in the rule body.
+          </p>
+          <p>
+            A [=rule=] is a <dfn>well-formed rule</dfn> if all the following
+            conditions are met:
+          </p>
+          <ul>
+            <li>For every [=variable=] appearing in a [=triple templates=]
+              of the [=head=] of the [=rule=],
+              there is one or more occurrences of a [=variable=] 
+              of the same name in the [=triple patterns=] in the [=body=], 
+              or in an [=assignment=] in the body, or both.
```

Comment:
```suggestion
            <li>For every [=variable=] appearing in a [=triple template=]
              of the [=head=] of the [=rule=],
              a [=variable=] 
              of the same name occurs one or more times in the
              [=triple patterns=] and/or the [=assignments=] in the body.
```

Reply:
I prefer to talk about "occurrences" at this point.